frontend/app/components/date-picker-field.tsx

```diff
@@ -0,0 +1,428 @@
+            {errorMessages.month}
+          </InputError>
+        ) : undefined,
+      day:
+        typeof errorMessages?.day === 'string' ? (
+          <InputError id={inputErrorIdDay} data-testid="date-picker-error-day">
+            {errorMessages.day}
+          </InputError>
+        ) : undefined,
+      year:
+        typeof errorMessages?.year === 'string' ? (
+          <InputError id={inputErrorIdYear} data-testid="date-picker-error-year">
+            {errorMessages.year}
+          </InputError>
+        ) : undefined,
+    };
+  }, [
+    errorMessages?.all,
+    errorMessages?.day,
+    errorMessages?.month,
+    errorMessages?.year,
+    inputErrorIdAll,
+    inputErrorIdDay,
+    inputErrorIdMonth,
+    inputErrorIdYear,
+  ]);
+
+  return (
+    <div id={inputWrapperId} data-testid="date-picker-field">
+      <fieldset>
+        <InputLegend id={inputLegendId} className="mb-2">
+          {legend}
+        </InputLegend>
+        {(datePickerErrorMessages.all !== undefined ||
+          datePickerErrorMessages.year !== undefined ||
+          datePickerErrorMessages.month !== undefined ||
+          datePickerErrorMessages.day !== undefined) && (
+          <div className="mb-2 space-y-2">
+            {datePickerErrorMessages.all && <p>{datePickerErrorMessages.all}</p>}
+            {currentLanguage === 'fr' ? (
+              <>
+                {datePickerErrorMessages.day && <p>{datePickerErrorMessages.day}</p>}
+                {datePickerErrorMessages.month && <p>{datePickerErrorMessages.month}</p>}
+                {datePickerErrorMessages.year && <p>{datePickerErrorMessages.year}</p>}
+              </>
+            ) : (
+              <>
+                {datePickerErrorMessages.month && <p>{datePickerErrorMessages.month}</p>}
+                {datePickerErrorMessages.day && <p>{datePickerErrorMessages.day}</p>}
+                {datePickerErrorMessages.year && <p>{datePickerErrorMessages.year}</p>}
+              </>
+            )}
+          </div>
+        )}
+        {helpMessagePrimary && (
+          <InputHelp
+            id={inputHelpMessagePrimaryId}
+            className={cn('mb-2', helpMessagePrimaryClassName)}
+            data-testid="date-picker-help-primary"
+          >
+            {helpMessagePrimary}
+          </InputHelp>
+        )}
+        <div className="flex flex-col space-y-2 sm:flex-row sm:space-y-0 sm:space-x-2">
+          {currentLanguage === 'fr' ? (
+            <>
+              {datePickerDay}
+              {datePickerMonth}
+              {datePickerYear}
+            </>
+          ) : (
+            <>
+              {datePickerMonth}
+              {datePickerDay}
+              {datePickerYear}
+            </>
+          )}
+        </div>
+        {helpMessageSecondary && (
+          <InputHelp
+            id={inputHelpMessageSecondaryId}
+            className={cn('mt-2', helpMessageSecondaryClassName)}
+            data-testid="date-picker-help-secondary"
```

Comment:
This will collide with other `data-testid` attributes if multiple date pickers are rendered in the same document.